src/lazy/text/buffer.rs

```diff
@@ -1177,8 +1177,49 @@ impl<'top> TextBuffer<'top> {
         Ok((exp_body_after_id, id))
     }
 
+    pub fn match_system_eexp_id(self) -> IonParseResult<'top, MacroIdRef<'top>> {
```

Comment:
🗺️ Text parsing logic for `$ion::<name>` and `$ion::<addr>`. Later on we can generalize this for any qualified macro reference.